### ISSUE TITLE
7.10 no longer accepts boost mapping on some properties

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/elasticsearch#64982]
```json
PUT /index-a


PUT /index-a/_mapping
{"properties":{"numberOfCommits":{"boost":1.5,"coerce":true,"ignore_malformed":true,"index":false,"null_value":0.0,"doc_values":true,"similarity":"BM25","store":true,"type":"integer"}}}
```

on `7.9.0` returns: 

```json
{
  "acknowledged" : true
}
```

In `7.10.0`

```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "mapper_parsing_exception",
        "reason" : "unknown parameter [boost] on mapper [numberOfCommits] of type [integer]"
      }
    ],
    "type" : "mapper_parsing_exception",
    "reason" : "unknown parameter [boost] on mapper [numberOfCommits] of type [integer]"
  },
  "status" : 400
}
```

This happens for all number type mappings, ip and token_count property mappings.

**Comment by mayya-sharipova** [comment]
`boost` is still accepted on `text` and `keyword` fields, but is no longer accepted on numeric types

**Comment by romseygeek** [comment]
This is definitely an error, you should get a deprecation message but not an error

**Comment by romseygeek** [comment]
Fixed by #64999

**Comment by RicardoGralhoz** [comment]
Maybe related to this topic: [Ignore_malformed fires mapper_parsing_exception for keyword type on 7.10.1 (it worked on 7.1.0)](https://discuss.elastic.co/t/ignore-malformed-fires-mapper-parsing-exception-for-keyword-type-on-7-10-1-it-worked-on-7-1-0/259142)
